package ec.edu.utm.Help_Desk_api.model;

import jakarta.persistence.*;
import java.util.List;

@Entity
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nombre;

    private String correo;

    private String rol;


    @OneToMany(mappedBy = "usuario")
    private List<Ticket> tickets;


    public Usuario(){

    }


    public Long getId() {
        return id;
    }


    public void setId(Long id) {
        this.id = id;
    }


    public String getNombre() {
        return nombre;
    }


    public void setNombre(String nombre) {
        this.nombre = nombre;
    }


    public String getCorreo() {
        return correo;
    }


    public void setCorreo(String correo) {
        this.correo = correo;
    }


    public String getRol() {
        return rol;
    }


    public void setRol(String rol) {
        this.rol = rol;
    }


    public List<Ticket> getTickets() {
        return tickets;
    }


    public void setTickets(List<Ticket> tickets) {
        this.tickets = tickets;
    }
}